functions

```diff
@@ -310,10 +312,10 @@ all_modules() {
     #   -f FILTER: ERE to filter found modules
 
     local -i count=0
-    local mod= OPTIND= OPTARG= filter=()
+    local mod='' OPTIND='' OPTARG='' filter=()
 
     while getopts ':f:' flag; do
-        case $flag in f) filter+=("$OPTARG") ;; esac
+        case "$flag" in f) filter+=("$OPTARG") ;; esac
```

Comment:
```suggestion
        [[ "$flag" = "f" ]] && filter+=("$OPTARG")
```

Reply:
The way it's written makes me think that it's intentional, but besides future proofing in case another parameter is added, I can't imaging why. :thinking: